test/e2e/tests/test_adopted_resource.py

```diff
@@ -0,0 +1,198 @@
+        resource_name=target_resource_name,
+        spec_file="sagemaker_endpoint_adopted_target",
+        replacements=replacements,
+    )
+
+    assert adopted_target_resource is not None
+
+    yield (resource_id, adopted_target_reference)
+
+    if k8s.get_resource_exists(adopted_target_reference):
+        _, deleted = k8s.delete_custom_resource(adopted_target_reference)
+        assert deleted
+
+
+@pytest.fixture(scope="module")
+def adopt_scaling_policy(adopt_scalable_target):
+    resource_id, adopted_target_reference = adopt_scalable_target
+    policy_resource_name = random_suffix_name("sagemaker-scaling-policy", 32)
+
+    replacements = REPLACEMENT_VALUES.copy()
+    replacements["ADOPTED_POLICY_NAME"] = policy_resource_name
+    replacements["RESOURCE_ID"] = resource_id
+
+    adopted_policy_reference, _, adopted_policy_resource = create_adopted_resource(
+        resource_name=policy_resource_name,
+        spec_file="sagemaker_endpoint_adopted_policy",
+        replacements=replacements,
+    )
+
+    assert adopted_policy_resource is not None
+
+    yield (resource_id, adopted_target_reference, adopted_policy_reference)
+
+    if k8s.get_resource_exists(adopted_policy_reference):
+        _, deleted = k8s.delete_custom_resource(adopted_policy_reference)
+        assert deleted
+
+
+@service_marker
+@pytest.mark.canary
+class TestAdoptedSageMakerEndpointAutoscaling:
+    def test_smoke(self, put_scaling_policy, adopt_scaling_policy):
+        sdk_resource_id = put_scaling_policy
+
+        (   adopted_resource_id,
+            adopted_target_reference,
+            adopted_policy_reference,
+        ) = adopt_scaling_policy
+
+        assert sdk_resource_id == adopted_resource_id
+
+        target_name = k8s.get_resource(adopted_target_reference)["spec"]["kubernetes"]["metadata"]["name"]
+        policy_name = k8s.get_resource(adopted_policy_reference)["spec"]["kubernetes"]["metadata"]["name"]
+
+        assert target_name is not None
+        assert policy_name is not None
+
+        for reference in (
+            adopted_target_reference,
+            adopted_policy_reference,
+        ):
+            assert k8s.wait_on_condition(reference, "ACK.Adopted", "True")
+
+        target_reference = k8s.create_reference(
+            CRD_GROUP, CRD_VERSION, TARGET_RESOURCE_PLURAL, target_name, "default"
+        )
+        target_resource = k8s.wait_resource_consumed_by_controller(target_reference)
+        assert target_resource is not None
+
+        assert target_resource["spec"].get("resourceID", None) == sdk_resource_id
+
+        policy_reference = k8s.create_reference(
+            CRD_GROUP, CRD_VERSION, POLICY_RESOURCE_PLURAL, policy_name, "default"
+        )
+        policy_resource = k8s.wait_resource_consumed_by_controller(policy_reference)
+        assert policy_resource is not None
+
+        assert policy_resource["spec"].get("resourceID", None) == sdk_resource_id
```

Comment:
@surajkota This assertion is there and IT WAS failing for me locally - it started passing on prow and thus I removed the WIP from this PR and assumed all was ok!

Reply:
It would work on Prow because Prow has no other ScalableTargets in the account (or shouldn't). Therefore the result of any ReadMany that doesn't specify `ResourceIds` should contain only one element - the scalable target you created in the fixture. It didn't work in your personal account because you may have had more than one scalable target already.

Reply:
yep, exactly.